package/endpoint_adapter/src/endpoint_adapter_can.c

```diff
@@ -10,17 +10,189 @@
  * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
  */
 
+#define _GNU_SOURCE
```

Comment:
What requires this? I have seen comments that this is a hit for portability.

Reply:
The `pipe2` call with O_DIRECT which enables packet style writes/reads from the pipe.